csrc/quantization/swizzled_quantize.cu

```diff
@@ -194,3 +195,236 @@ void launch_swizzled_quant(int8_t* q_data,
         }
     }
 }
+
+template <int numBits, int totalChunks, int threads, quantize::Type quantType>
+__global__ void loco_swizzled_quant_kernel(int8_t* quantized_data,
+                                           float* quantized_scales,
+                                           const __half* uncompressed_data,
+                                           __half* error_feedback,
+                                           const float err_beta,
+                                           int groups,
+                                           int elems_per_group,
+                                           int pipelining,
+                                           int nodes,
+                                           int devices_per_node)
+{
+    cg::thread_block tb = cg::this_thread_block();
+    cg::thread_block_tile<hw_warp_size> warp = cg::tiled_partition<hw_warp_size>(tb);
+
+    int blockIdx_x = blockIdx.x;
+    int blockIdx_y = blockIdx.y;
+    int blockIdx_z_swizzled = blockIdx.z;
+
+    int device_id = blockIdx_z_swizzled / nodes;
+    int node_id = blockIdx_z_swizzled % nodes;
+
+    int blockIdx_z_orig = node_id * devices_per_node + device_id;
+
+    int block_rank_orig =
+        blockIdx_x + blockIdx_y * gridDim.x + blockIdx_z_orig * gridDim.x * gridDim.y;
+
+    const int elem_offset = threadIdx.x * quantize::h_per_load;
+    const int block_offset_orig = block_rank_orig * elems_per_group;
+    const int base_offset_orig = block_offset_orig + elem_offset;
+    const int stride = blockDim.x * quantize::h_per_load;
+    const __half* uncompressed_data_base = uncompressed_data + base_offset_orig;
+
+    const int block_rank_swizzled =
+        blockIdx_x + blockIdx_y * gridDim.x + blockIdx_z_swizzled * gridDim.x * gridDim.y;
+    const int block_offset_swizzled = block_rank_swizzled * elems_per_group;
+    const int base_offset_swizzled = block_offset_swizzled + elem_offset;
+    __half* error_feedback_base = error_feedback + base_offset_swizzled;
+
+    __half2 local_buffer[totalChunks * quantize::h_per_load];
+    __half2 err_buffer[totalChunks * quantize::h_per_load];
```

Comment:
You use both `quantize::h_per_load` and `quantize::h2_per_load` in this kernel code. Can you double check to make sure that both of them were used correctly? This one doesn't match the code line 245 and 246.